presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/TableMetadataPageSource.java

```diff
@@ -0,0 +1,410 @@
+
+        Iterator<TableRow> tableRows = dao.getTables(schemaName, tableName).iterator();
+        PeekingIterator<ColumnRow> columnRowIterator = Iterators.peekingIterator(dao.getColumns(schemaName, tableName).iterator());
+
+        while (tableRows.hasNext()) {
+            TableRow tableRow = tableRows.next();
+            while (columnRowIterator.hasNext() && columnRowIterator.peek().getTableId() < tableRow.getTableId()) {
+                columnRowIterator.next();
+            }
+
+            String temporalColumnName = null;
+            ArrayList<String> sortColumnNames = new ArrayList<>();
+            ArrayList<String> bucketColumnNames = new ArrayList<>();
+            while (columnRowIterator.hasNext() && columnRowIterator.peek().getTableId() == tableRow.getTableId()) {
+                ColumnRow columnRow = columnRowIterator.next();
+                if (tableRow.getTemporalColumnId().isPresent() && columnRow.getColumnId() == tableRow.getTemporalColumnId().getAsLong()) {
+                    temporalColumnName = columnRow.getColumnName();
+                }
+                if (columnRow.getSortOrdinalPosition().isPresent()) {
+                    int sortOrdinalPosition = columnRow.getSortOrdinalPosition().getAsInt();
+                    ensureSize(sortColumnNames, sortOrdinalPosition + 1);
+                    sortColumnNames.set(sortOrdinalPosition, columnRow.getColumnName());
+                }
+                if (columnRow.getBucketOrdinalPosition().isPresent()) {
+                    int bucketOrdinalPosition = columnRow.getBucketOrdinalPosition().getAsInt();
+                    ensureSize(bucketColumnNames, bucketOrdinalPosition + 1);
+                    bucketColumnNames.set(bucketOrdinalPosition, columnRow.getColumnName());
+                }
+            }
+
+            if (sortColumnNames.stream().anyMatch(sortColumnName -> sortColumnName == null)) {
+                throw new VerifyException(
+                        format("sortColumnNames has null: tableId %s, schemaName %s, tableName %s, sortColumnNames %s", tableRow.getTableId(), tableRow.getSchemaName(), tableRow.getTableName(), sortColumnNames.toString()));
+            }
+            if (bucketColumnNames.stream().anyMatch(bucketColumnName -> bucketColumnName == null)) {
+                throw new VerifyException(
+                        format("bucketColumnNames has null: tableId %s, schemaName %s, tableName %s, bucketColumnNames %s", tableRow.getTableId(), tableRow.getSchemaName(), tableRow.getTableName(), bucketColumnNames.toString()));
+            }
+
+            pageBuilder.declarePosition();
+            VARCHAR.writeSlice(pageBuilder.getBlockBuilder(0), utf8Slice(tableRow.getSchemaName()));
+            VARCHAR.writeSlice(pageBuilder.getBlockBuilder(1), utf8Slice(tableRow.getTableName()));
+            if (tableRow.getTemporalColumnId().isPresent()) {
+                if (temporalColumnName == null) {
```

Comment:
Can this only happen due to a code bug, or would corrupt metadata also cause this? If the latter, make this throw `PrestoException(RAPTOR_ERROR)`